### PR TITLE
Add Smarts Strings for Sulfur Compounds

### PR DESCRIPTION
**Pull request by Argon1999**
### PR Summary:
Added Smarts strings for Disulfides, H2S, Sulfides and Thiols. From OPLS_200 to OPLS 217.
Encountering a problem with atom-typing thiols with opls_207. With current implementation Foyer is unable to atom-type the carbon in 2R-CH-S-H. I have checked override errors and the Smarts strings looks to be accurate. One possible reason is that foyer wants the R's to be specified as Carbons or something else. This should get fixed before merging.

### PR Checklist
------------
 - [x] Includes appropriate unit test(s)
 - [x] Appropriate docstring(s) are added/updated
 - [x] Code is (approximately) PEP8 compliant
 - [ ] Issue(s) raised/addressed?

### CODE REVIEW

File: foyer/opls_validation/dimethyldisulfide/dimethyldisulfide.gro [hunk 1]
```diff
@@ -0,0 +1,103 @@
+GROningen MAchine for Chemical Simulation
+  100
```
Comment (rmatsum836) [inline foyer/opls_validation/dimethyldisulfide/dimethyldisulfide.gro]: Can you update all these gro files to only have 1 molecule each?

Comment (Argon1999) [reply foyer/opls_validation/dimethyldisulfide/dimethyldisulfide.gro]: Sure I'll do it now.

**Comment by justinGilmer** [comment]
After tests pass again, lgtm! Thanks @Argon1999 !

**Comment by mattwthompson** [comment]
This is good to merge but I am curious if you two found a fix for that issue that stumped us a few weeks ago? @rmatsum836 @Argon1999

**Comment by Argon1999** [comment]
@mattwthompson I did not. I just removed it.

**Comment by mattwthompson** [comment]
Weird. I will merge this but I would appreciate it if you could write up an explanation of the error, so that it's saved somewhere publicly for posterity @Argon1999 A comment in this thread is fine

File: foyer/forcefields/xml/oplsaa.xml [hunk 2]
```diff
@@ -196,24 +196,24 @@
...
-  <Type name="opls_212" class="opls_212" element="C" mass="12.011"/>
-  <Type name="opls_213" class="opls_213" element="C" mass="12.011"/>
-  <Type name="opls_214" class="opls_214" element="C" mass="12.011"/>
-  <Type name="opls_215" class="opls_215" element="C" mass="12.011"/>
-  <Type name="opls_216" class="opls_216" element="C" mass="12.011"/>
-  <Type name="opls_217" class="opls_217" element="C" mass="12.011"/>
+  <Type name="opls_200" class="SH" element="S" mass="32.06" def="[S;X2]H" desc="S in any Thiol" overrides="opls_202" doi="10.1021/ja9621760"/>
+  <Type name="opls_201" class="SH" element="S" mass="32.06" def="[S;X2](H)(H)" desc="S in H2S" overrides="opls_200, opls_202" doi="10.1021/ja9621760"/>
+  <Type name="opls_202" class="S" element="S" mass="32.06" def="[S;X2]" desc="S in any sulfide" doi="10.1021/ja9621760" />
+  <Type name="opls_203" class="S" element="S" mass="32.06" def="[S;X2](S)(*)" desc="S in Disulfide" overrides="opls_200, opls_202" doi="10.1021/ja9621760"/>
```
Comment (mattwthompson) [inline foyer/forcefields/xml/oplsaa.xml]:
```suggestion
  <Type name="opls_203" class="S" element="S" mass="32.06" def="[S;X2](S)" desc="S in Disulfide" overrides="opls_200, opls_202" doi="10.1021/ja9621760"/>
```

AFAICT the wildcard does nothing?

**Pull request merged**